### PR TITLE
Upgrade to Babel 6

### PR DESCRIPTION
**Pull request by marvinroger**
Everything is working perfectly. Code coverage is also fixed.

### CODE REVIEW

File: flyfile.js [hunk 1]
```diff
@@ -10,6 +10,6 @@ export default function* () {
   yield this.clear(paths.dist)
   yield this.log("Building Fly...")
     .source(paths.src)
-    .babel({ optional: ["runtime"], modules: "common" })
+    .babel({ presets: ["es2015", "stage-3"] })
```
Comment (marvinroger) [inline flyfile.js:13]: `es2015`: ES6 support
`stage-3`: support for ES7 async/await

File: src/cli/index.js [hunk 2]
```diff
@@ -1,5 +1,5 @@
-export * from "./help"
-export * from "./list"
-export * from "./version"
-export * from "./spawn"
-export * from "./options"
+export {help} from "./help"
```
Comment (marvinroger) [inline src/cli/index.js:1]: Workaround for the `export * from "..."` issue in Babel 6. See https://phabricator.babeljs.io/T2763

File: test/reporter.js [hunk 3]
```diff
@@ -26,7 +26,7 @@ test("✈  reporter", (t) => {
 })
 
 test("✈  timeInfo", (t) => {
-  const timeInfo = require("../src/reporter/timeInfo")
+  const timeInfo = require("../src/reporter/timeInfo").default
```
Comment (marvinroger) [inline test/reporter.js:29]: In Babel 6, the `default` is not exported at top level when using `require` for security reason. We have to explicitely add `.default`. [See here](http://webcache.googleusercontent.com/search?q=cache:i9V50YCRtT0J:github.com.proxy.parle.co/babel/babel/issues/2212+&cd=3&hl=fr&ct=clnk&gl=fr) (sorry, this is only available from Google cache). There is [a plugin](https://www.npmjs.com/package/babel-plugin-add-module-exports) to patch that, but obviously [there is also a bug](https://phabricator.babeljs.io/T6644) that make plugins conflicting. :+1: It's not that problematic, as this is only affecting code `require`'ing ES6 exports. ES6 import works fine.

File: package.json [hunk 4]
```diff
@@ -46,13 +48,13 @@
     "symlink": "ln -s $PWD/bin/index /usr/local/bin/fly",
     "clean": "rimraf dist",
     "lint": "eslint src test",
-    "coverage": "babel-node node_modules/.bin/isparta cover test/*.js",
+    "coverage": "babel-node --presets es2015,stage-3 ./node_modules/.bin/isparta cover test/*.js",
```
Comment (marvinroger) [inline package.json]: Code coverage still doesn't work, but I can't help on that. With the current published version, it doesn't work either, and I don't know how isparta/istanbul work. Sorry!

Comment (iamstarkov) [inline package.json]: @marvinroger @bucaran take a look at latest change in [tiny-es-nm generator](https://github.com/iamstarkov/generator-tiny-es-nm) i finally got coverage working with babel v6. It’s here https://github.com/iamstarkov/generator-tiny-es-nm/blob/master/app/templates/_package.json#L8

Comment (marvinroger) [inline package.json]: @iamstarkov thanks, but you use the `--compilers` option of mocha. Fly uses tape.

Comment (marvinroger) [inline package.json]: Adding a .babelrc did the trick! Coverage is fixed! :)

Comment (iamstarkov) [inline package.json]: i told ya!

File: flyfile.js (hunk 1 above)
Comment (ghost) [inline flyfile.js:13]: :+1:

File: src/cli/index.js (hunk 2 above)
Comment (ghost) [inline src/cli/index.js:1]: I was too excited about modules that I left myself carried away with this stuff. :+1:

File: test/reporter.js (hunk 3 above)
Comment (ghost) [inline test/reporter.js:29]: Well, we can live with it for a while right?

**Comment by ghost** [comment]
Thanks @marvinroger :+1:

**Pull request merged**